pkg/services/ngalert/api/api_convert_prometheus.go

```diff
@@ -377,6 +389,116 @@ func (srv *ConvertPrometheusSrv) RouteConvertPrometheusPostRuleGroup(c *contextm
 	return successfulResponse()
 }
 
+func (srv *ConvertPrometheusSrv) RouteConvertPrometheusPostDatasource(c *contextmodel.ReqContext, datasourceUID string) response.Response {
+	logger := srv.logger.FromContext(c.Req.Context())
+
+	pauseRecordingRules := c.QueryBoolWithDefault("pauseRecordingRules", true)
+	pauseAlerts := c.QueryBoolWithDefault("pauseAlerts", true)
+	targetFolderUID := c.Query("targetFolderUID")
+	namespaceFilter := c.Query("namespace")
+	groupFilter := c.Query("group")
+
+	// 1. Fetch rules from datasource
+	promGroups := map[string][]prom.PrometheusRuleGroup{}
+	resp := srv.proxySvc.requester.withReq(c, http.MethodGet, withPath(*c.Req.URL, MimirPrefix), nil, yamlExtractor(&promGroups), nil)
+
+	// 2. Convert Prometheus Rules to GMA
+	if err := json.Unmarshal(resp.Body(), &promGroups); err != nil {
+		return errorToResponse(err)
+	}
+
+	grafanaGroups := make([]*models.AlertRuleGroup, 0, len(promGroups))
+	for ns, rgs := range promGroups {
+		// Check Namespace filter
+		if namespaceFilter != "" && ns != namespaceFilter {
+			continue
+		}
+
+		// Need to filter groups here so we don't end up unnecessarily creating empty namespaces
+		if groupFilter != "" {
+			filteredRgs := make([]prom.PrometheusRuleGroup, 0, len(rgs))
+			for _, rg := range rgs {
+				if rg.Name != groupFilter {
+					continue
+				}
+				filteredRgs = append(filteredRgs, rg)
+			}
+			rgs = filteredRgs
+		}
+
+		logger.Debug("Creating a new namespace", "title", ns)
+		namespace, errResp := srv.getOrCreateNamespace(c, ns, logger, targetFolderUID)
```

Comment:
Something that @gillesdemey pointed out: it's currently not possible not to specify `targetFolderUID`, but it should be optional, maybe `getWorkingFolderUID` can be used here